File: golang/euler/2.go
package main

import "fmt"

func main(){
    var fibs []int
    fibs = append(fibs, 1)
    fibs = append(fibs, 2)
    till := 4000000
    i := 0
    for fibs[i+1] + fibs[i] < till{
        fibs = append(fibs, fibs[i] + fibs[i+1])
        i += 1
    }

    fmt.Println(fibs)


    sum := 0
    for _, value := range fibs {
        if value % 2 ==0{
            sum += value
        }

    }
    fmt.Println(sum)

}

File: julia/euler/2.jl
function is_even(num)
    return (num % 2) == 0
end

x, y = 1, 2
fibs = [1]
last = 4000000
while y < last
  push!(fibs, y)
  x, y = y, x + y
end
fibs = filter(is_even, fibs)
println(sum(fibs))
